addon/PrefsProvider.js

```diff
@@ -13,51 +13,37 @@ exports.PrefsProvider = class PrefsProvider {
    *
    * @param  {obj} options
    *         {func} options.broadcast    This is a method that takes an action created with am.actions (action-manager.js)
-   *         {func} options.send         This is a method that takes an action created with am.actions, and a worker.
-   *                                     Note: broadcast and send are defined on the ActivityStreams.js instance.
    */
-  constructor(options) {
-    const {broadcast, send} = options;
-    this.broadcast = broadcast;
-    this.send = send;
+  constructor(tabTracker, options) {
+    this.broadcast = options.broadcast;
+    this._tabTracker = tabTracker;
+    this._onPrefChange = this._onPrefChange.bind(this);
   }
 
   /**
-   * init - Sets up a listener on pref changes that broadcasts the change to content
-   *        in the form {name: "prefName", value: "newPrefvalue"}
+   * Broadcast the pref change to content in the form
+   *  {name: "prefName", value: "newPrefvalue"} and capture that event
    */
-  init() {
-    this.onPrefChange = name => this.broadcast(am.actions.Response("PREF_CHANGED_RESPONSE", {
+  _onPrefChange(name) {
+    this.broadcast(am.actions.Response("PREF_CHANGED_RESPONSE", {
       name,
       value: simplePrefs.prefs[name]
     }));
-    simplePrefs.on("", this.onPrefChange);
+    this._tabTracker.handleUserEvent({"event": "PREF_CHANGE", "source": name});
   }
 
   /**
-   * destroy - Removes the event listener on prefs
+   * init - Sets up a listener on pref changes
    */
-  destroy() {
-    simplePrefs.off("", this.onPrefChange);
-    this.onPrefChange = null;
+  init() {
+    simplePrefs.on("", this._onPrefChange);
   }
 
   /**
-  * actionHandler - This is added to the message channel in ActivityStreams.js
-  *                 It handles messages from content.
-   * @param  {obj} options
-   *         {obj} options.msg     This is the message from content (i.e. the 'action')
-   *         {obj} options.worker  This is the worker to which this.send responds
+   * destroy - Removes the event listener on prefs
    */
-  actionHandler(options = {}) {
-    const {msg: action, worker} = options;
-    switch (action.type) {
-      case "PREFS_REQUEST":
-        this.send(am.actions.Response("PREFS_RESPONSE", simplePrefs.prefs), worker);
```

Comment:
Yep, that sounds right. We were never hitting this action manager anyways since we were sending PREFS_RESPONSE down by refreshAppState